maxim/models/dataset.py

```
@@ -9,128 +9,215 @@ class VariableType(str):
-            return_dict["context"] = {
-                "type": self.context.type,
-                "payload": self.context.payload,
-            }
-        if self.expectedOutput is not None:
-            return_dict["expectedOutput"] = {
-                "type": self.expectedOutput.type,
-                "payload": self.expectedOutput.payload,
-            }
-        return return_dict
+        if not isinstance(data_dict, dict):
+            raise ValueError("Input must be a dictionary")
```

Ruff (0.12.2)

Prefer `TypeError` exception for invalid type

(TRY004)